Cython/Compiler/Symtab.py

```diff
@@ -922,20 +923,44 @@ def find_imported_module(self, path, pos):
     def lookup(self, name):
         # Look up name in this scope or an enclosing one.
         # Return None if not found.
-        name = self.mangle_class_private_name(name)
-        return (self.lookup_here(name)
-            or (self.outer_scope and self.outer_scope.lookup(name))
-            or None)
+
+        mangled_name = self.mangle_class_private_name(name)
+        entry = (self.lookup_here(name)  # lookup here also does mangling
+                or (self.outer_scope and self.outer_scope.lookup(mangled_name))
+                or None)
+        if entry:
+            return entry
+
+        # look up the original name in the outer scope
```

Comment:
Hmm, this is still done in the base class. Why not in `ClassScope`?

Reply:
I don't think it can be. `mangle_class_private_name` mangling uses `parent_scope` (so in a typical case will start in a function local scope, then go up to class scope. `lookup` uses `outer_scope`, so would typically jump from a function local scope to a module scope (and not involve the `ClassScope`)